src/server/pullJobScheduler.ts

```diff
@@ -347,6 +352,58 @@ export const insertRecords = async (
       form: pullJob.convertTo,
       $or: filters,
     }).select(selectedFields);
+
+    // If EIOS pullJob, build a mapping JSON to assign ownership (role ids)
+    const ownershipMappingWithIds: any = {};
+    if (isEIOS) {
+      // Get Id of Signal HQ PHI app to assign to every element
+      const applicationSignalHQPHI = await Application.find({
+        name: 'Signal HQ PHI',
+      });
+      const signalHQPHIUserRole = await Role.findOne({
+        application: applicationSignalHQPHI[0]._id,
+        title: 'User',
+      });
+      for (const [key, value] of Object.entries(ownershipMappingJSON)) {
+        ownershipMappingWithIds[key] = [];
+        if (value.length > 0) {
+          //Create a filter to get roles based on application name(s)
+          const filterList: any[] = [
+            {
+              $lookup: {
+                from: 'applications',
+                localField: 'application',
+                foreignField: '_id',
+                as: '_application',
+              },
+            },
+          ];
+          const regionFilters = [];
+          // Mandatory to use map function
+          const regionArrays = value as string[];
+          regionArrays.map((elt) => {
+            regionFilters.push({
+              $and: [
+                { title: 'User' },
+                { _application: { $elemMatch: { name: { $regex: elt } } } },
```

Comment:
The regex is risky I think in case new apps are created containing regions names (not unlikely). I think we should replace the Regions names by the complete application name in the mapping and do a full match at this step.